src/target/linux/memory.rs

```diff
@@ -0,0 +1,36 @@
+//! Utility functions to work with memory.
+
+use crate::target::MemoryMap;
+use std::cmp::Ordering;
+
+lazy_static::lazy_static! {
+    /// Memory page size from system configuration.
+    pub(crate) static ref PAGE_SIZE: usize = unsafe { libc::sysconf(libc::_SC_PAGESIZE) as usize };
+}
+
+/// Abstract memory operation (reading or writing).
+pub trait MemoryOp {
+    /// Returns a remote address at which this memory operation will be applied to.
+    fn remote_base(&self) -> usize;
+}
+
+/// Splits memory operations to those that read from read protected memory and those that do not.
+pub fn split_protected<'a, M: MemoryOp + Clone>(
+    maps: &'a [MemoryMap],
+    operations: &'a [M],
+) -> Result<(Vec<M>, Vec<M>), Box<dyn std::error::Error>> {
+    let (protected, readable): (_, Vec<_>) = operations.iter().cloned().partition(|op| {
+        maps.binary_search_by(|map| {
+            if op.remote_base() < map.address.0 as usize {
+                Ordering::Greater
+            } else if op.remote_base() > map.address.1 as usize {
+                Ordering::Less
+            } else {
+                Ordering::Equal
+            }
+        })
+        .is_ok()
```

Comment:
This should check `is_readable`.

Reply:
It does, but in an implicit way - `maps` are pre-filtered as `memory_maps().into_iter().filter(|m| !m.is_readable())` at the call site. I'm not sure if we should make it a func argument for explicit filtering, or leave it as a comment suggesting that it's a user's responsibility.

Reply:
Please take an `Iterator` instead of slice then to avoid an unnecessary allocation and explain in the doc comment that the caller is responsible for filtering.

Reply:
I actually meant for `maps` to be an iterator.